### ISSUE TITLE
[Google Local API] No results for valid search

### ISSUE THREAD

**Issue by marm123** [serpapi/public-roadmap#2166]
One of our high-volume customers reported the issue with Google Local. For some searches, they are not getting any results, and no HTML is rendered at all.

<img width="1138" alt="image" src="https://github.com/user-attachments/assets/891f5981-6dbd-4c88-a188-395baf08fd4e">

This looks to be related to other issue fixed with https://github.com/serpapi/SerpApi/pull/5536. @dsojevic can you take a look and see if it's the same issue?

Related Public Roadmap issue: https://github.com/serpapi/public-roadmap/issues/2147

Internal links: [Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000072412?view=List) | [Search Inspect 1](https://serpapi.com/searches/673b939296a3296332b6f433/inspect) | [Search Inspect 2](https://serpapi.com/searches/673b938f8f2bc25994b8afb5/inspect)

**Comment by dsojevic** [comment]
@marm123, I've just digging into this and have been unable to reproduce it.

It appears as though the example searches were from prior to https://github.com/serpapi/SerpApi/pull/5536 being rolled out so all searches on Google Local from ~2024-11-20 onwards should not be exhibiting any issues.

**Comment by schaferyan** [comment]
I'm re-opening this issue as the customer who reported this provided some recent examples:

[Inspect](https://serpapi.com/searches/6744aa1cd364b476f5f0e691/inspect) | [Inspect 2](https://serpapi.com/searches/674493b56273fb8e290c34fe/inspect) | [Inspect 3](https://serpapi.com/searches/6744ab8cacf7f8e200a0b3fa/inspect)

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000072412?view=List)

**Comment by dsojevic** [comment]
> I'm re-opening this issue as the customer who reported this provided some recent examples:

Thanks @schaferyan. I'm going to have to investigate further; unfortunately these new searches are affected by something unrelated to the issue that caused the original searches to fail/return empty.

**Comment by dsojevic** [comment]
@schaferyan @marm123, I've spent a fair bit of time digging into this and have been unable to find anything that would indicate this type of response back from the Google search might be something we can address on our side (i.e. Google is returning a "successful" response and they are indicating that they have no results for the query)

I've done large batches of testing with various parameters/setups to see if it perhaps prevalent with any particular combination and haven't encountered any issues in ~1,500 searches worth of testing.

I ran a final batch of testing with the same parameters seen in **Inspect 3** and came out with the following results which had results returned in 100% of requests:

| Category         | Number of Requests | Average Reported Time | Average Result Count |
|------------------|--------------------|------------------------|-----------------------|
| Overall          | 200                | 2.1658                | 20.0                 |
| Valid Requests   | 200                | 2.1658                | 20.0                 |
| Invalid Requests | 0                  | -                     | -                    |
| Failed Requests  | 0                  | -                     | -                    |

When a user is performing searches with queries that they are confident should be returning results, the path forward might be for them to do one _(or more, with a limit)_ retries when they encounter an empty result set.

I'm not sure there's much else we can do at this time until we're able to reproduce the issue on demand.

---

It might also be worth considering the implications of us passing along informational notices in the `error` key in our responses as I can see it being somewhat confusing. For example, here is a search for a UUID that you would expect to have no results returned:
![image](https://github.com/user-attachments/assets/e481dee3-f5ac-40ee-9c0c-221cc916fbc0)

**Comment by schaferyan** [comment]
Thanks, @dsojevic! I let the customer know.

> It might also be worth considering the implications of us passing along informational notices in the error key in our responses as I can see it being somewhat confusing. 

Do you mean this information should be included in a different key than `error`? I can confirm that this confuses customers. It's not really an error if the search engine itself returns empty results.

**Comment by dsojevic** [comment]
> Do you mean this information should be included in a different key than `error`? I can confirm that this confuses customers. It's not really an error if the search engine itself returns empty results.

Correct! Perhaps in this case it can even be fully omitted because the same information is also conveyed under `search_information.local_results_state` with `"Fully empty"`?

**Comment by schaferyan** [comment]
@dsojevic I think the goal of this message is to clarify that is the search engine itself that didn't return any results, not an issue on our side. Whereas `search_information.local_results_state: "Fully empty"` leaves more ambiguity whether it's a failure on our side or Google's. 

I agree though that providing it in `error` is counter to that purpose.

Maybe we should open a separate internal issue to discuss this?

**Comment by schaferyan** [comment]
Another customer reported this. I was able to replicate it with ~10 retries. They believe it may be related to omitting the country from the `location` parameter, but I was able to replicate the issue regardless of whether that is included.

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/215471641460506?view=List)
[Inspect](https://serpapi.com/searches/6914dcaeaa0a8876867452fd/inspect)

**Comment by jbcoventry** [comment]
I had this occur when investigating the above issue. Also on playground.

[Inspect](https://serpapi.com/searches/695cbc7b66bc78f1b82f00a4/inspect)

**Comment by ocvit** [comment]
The remaining issue related to Google Local has been resolved.

Hotels and other accommodations are parsed correctly now.

<img width="1679" height="582" alt="Image" src="https://github.com/user-attachments/assets/8336a885-2266-496f-bc46-4358c1881402" />
